internal/resources/kubernetessecret/resource_secret.go

```diff
@@ -34,52 +34,67 @@ func ResourceSecret() *schema.Resource {
 		DeleteContext: resourceSecretDelete,
 		UpdateContext: resourceSecretInPlaceUpdate,
 		ReadContext:   dataSourceSecretRead,
-		Schema:        secretSchema,
+		Schema:        getSecretSchema(false),
 	}
 }
 
-var secretSchema = map[string]*schema.Schema{
-	NameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of the secret resource.",
-		Required:    true,
-		ForceNew:    true,
-		ValidateFunc: validation.All(
-			validation.StringLenBetween(1, 126),
-			validation.StringIsNotEmpty,
-			validation.StringIsNotWhiteSpace,
-		),
-	},
-	NamespaceNameKey: {
-		Type:        schema.TypeString,
-		Description: "Name of Namespace where secret will be created.",
-		Required:    true,
-		ForceNew:    true,
-		ValidateFunc: validation.All(
-			validation.StringIsNotEmpty,
-			validation.StringIsNotWhiteSpace,
-		),
-	},
-	OrgIDKey: {
-		Type:        schema.TypeString,
-		Description: "ID of Organization.",
-		Optional:    true,
-	},
-	scope.ScopeKey: scope.ScopeSchema,
-	specKey:        secretSpec,
-	common.MetaKey: common.Meta,
-	ExportKey: {
-		Type:        schema.TypeBool,
-		Description: "Export the secret to all namespaces.",
-		Optional:    true,
-		Default:     false,
-	},
-	statusKey: {
-		Type:        schema.TypeMap,
-		Description: "Status for the Secret Export.",
-		Computed:    true,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-	},
+func getSecretSchema(isDataSource bool) map[string]*schema.Schema {
```

Comment:
Also add unit tests and ensure that the schema does not change

Reply:
added PTAL

Reply:
output

```
=== RUN   TestGetSecretSchema
=== RUN   TestGetSecretSchema/resource_schema_test
--- PASS: TestGetSecretSchema/resource_schema_test (0.00s)
=== RUN   TestGetSecretSchema/data_source_schema_test
--- PASS: TestGetSecretSchema/data_source_schema_test (0.00s)
=== RUN   TestGetSecretSchema/generate_wrong_resource_schema
--- PASS: TestGetSecretSchema/generate_wrong_resource_schema (0.00s)
--- PASS: TestGetSecretSchema (0.00s)
PASS
ok      github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/kubernetessecret  1.782s


> Test run finished at 12/04/2023, 18:49:23 <

```